jax/_src/lax/lax.py

```diff
@@ -3301,9 +3301,14 @@ def _broadcast_in_dim_shape_rule(operand, *, shape, broadcast_dimensions):
 
   return shape
 
-def _broadcast_in_dim_transpose_rule(t, *, shape, broadcast_dimensions):
-  axes = tuple(np.delete(range(len(shape)), broadcast_dimensions))
-  return [_reduce_sum(t, axes)]
+def _broadcast_in_dim_transpose_rule(ct, operand, *, shape, broadcast_dimensions):
+  if type(ct) is ad_util.Zero:
+    return ad_util.Zero
```

Comment:
We need to return an instance of `ad_util.Zero` here, not the class itself.

```suggestion
    return ad_util.Zero(operand.aval)
```

Reply:
I was following `linear_transpose`, which returns the class rather than an instance: https://github.com/google/jax/blob/19b3cf9a7222768aecfb606b54796367427c5fa8/jax/interpreters/ad.py#L440

Should that be changed as well?

Reply:
This is no longer relevant to this PR; I'll open a separate issue regarding returning the `Zero` class in `ad.py`.

Reply:
Good catch! Yeah, I think that's a bug. (We used to do something like that, where we had a singleton class for zero, but when we changed zeros to carry avals we should've replaced those things.)

I'm surprised it wasn't caught by [this assertion](https://github.com/google/jax/blob/62f943435f20345aa2cafd6e0201de77ae459352/jax/interpreters/ad.py#L169). We must not be hitting that `linear_transpose` case somehow...